### PR TITLE
Draft: Fix svg images not correctly displaying in e2ee rooms

### PR DESCRIPTION
**Pull request by justjanne**
Type: Defect
Related: https://github.com/vector-im/element-web/issues/15094
Related: https://github.com/vector-im/element-web/issues/18526

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Draft: Fix svg images not correctly displaying in e2ee rooms ([\#8325](https://github.com/matrix-org/matrix-react-sdk/pull/8325)). Contributed by @justjanne.

### CODE REVIEW

**Comment by justjanne** [comment]
This PR uses data URLs for embedded SVGs in E2EE rooms and other situations where no PNG version is available. This ensures we get a high quality preview without the risk of exposing our origin to user-defined scripts.

File: src/utils/MediaEventHelper.ts
Comment (turt2live) [inline src/utils/MediaEventHelper.ts:60]: these need return types, and a blank line between them

File: src/utils/MediaEventHelper.ts
Comment (turt2live) [inline src/utils/MediaEventHelper.ts:36]: these were deliberately exposed and should not be hidden. It's for overall performance of the app.

File: src/utils/MediaEventHelper.ts
Comment (turt2live) [inline src/utils/MediaEventHelper.ts:109]: `info` might not exist, and `decryptFile` should be responsible for mimetype detection.

File: src/utils/MediaEventHelper.ts
Comment (turt2live) [inline src/utils/MediaEventHelper.ts:122]: `info` might not exist, and `decryptFile` should be responsible for mimetype detection.

`thumbnail_info` also might not exist

File: src/utils/MediaEventHelper.ts
Comment (turt2live) [inline src/utils/MediaEventHelper.ts:43]: This opens us up to XSS attacks, which we do not want.

**Review by turt2live — CHANGES_REQUESTED** [review]
Thanks for taking a look at this, though it's not something we can merge easily. In addition to the comments below, I think this needs explicit review from the security team.

File: src/utils/MediaEventHelper.ts
Comment (turt2live) [reply src/utils/MediaEventHelper.ts:36]: (also used by projects downstream of us)

**Comment by justjanne** [comment]
@turt2live I appreciate the review, after all this is just a first attempt at experimenting with potential solutions that allow us to get SVG previews working.

I'll experiment further with this over the easter weekend, though I don't expect that I'll get a clean fix done soon.

(I actually thought I had left the PR in draft status, but that may have gotten lost at some point)

**Comment by turt2live** [comment]
ah, apologies for the early review then 😅 

This is a relatively security sensitive area though, particularly the object URL handling.

**Pull request closed without merging**
